core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -241,7 +242,29 @@ ByteBuffer readKey(int entryIndex) {
         int keyPosition = keyArray[POSITION_ARRAY_INDEX];
         int length = keyArray[BLOCK_ID_LENGTH_ARRAY_INDEX] & KEY_LENGTH_MASK;
 
-        return memoryManager.getByteBufferFromBlockID(blockID, keyPosition, length);
+        return memoryManager.getByteBufferFromBlockID(blockID, keyPosition, length,
+            OakNativeMemoryAllocator.FIRST_THREAD_BUFFER);
+    }
+
+    /**
+     * Reads a key given the entry index. Key is returned via reusable thread-local ByteBuffer.
+     * There is no copy just a special ByteBuffer for a single key.
+     * The thread-local ByteBuffer can be reused by different threads, however as long as
+     * a thread is invoked the ByteBuffer is related solely to this thread.
+     */
+    ByteBuffer readSecondKey(int entryIndex) {
+        if (entryIndex == Chunk.NONE) {
+            return null;
+        }
+
+        long keyReference = getKeyReference(entryIndex);
+        int[] keyArray = longToInts(keyReference);
+        int blockID = keyArray[BLOCK_ID_LENGTH_ARRAY_INDEX] >> KEY_BLOCK_SHIFT;
+        int keyPosition = keyArray[POSITION_ARRAY_INDEX];
+        int length = keyArray[BLOCK_ID_LENGTH_ARRAY_INDEX] & KEY_LENGTH_MASK;
```

Comment:
code duplication 256-264 should go through 3rd method that gets first/second as parameter